src/main/com/yetanalytics/lrs/pedestal/interceptor/xapi/statements.cljc

```diff
@@ -236,6 +236,33 @@
                 {:status 400
                  :body {:error {:message "No Statement Data Provided"}}}))))})
 
+(defn scan-attachments
+  "Scan attachment files with a user-provided function."
+  [file-scanner]
+  {:name ::scan-attachments
+   :enter
+   (fn [ctx]
+     (let [attachments (get-in ctx [:xapi :xapi.statements/attachments])]
+       (if-let [attachment-errors
+                (some-> attachments
+                        (->> (keep (fn [{:keys [content]}]
+                                     (try
+                                       (file-scanner content)
+                                       (catch #?(:clj Exception :cljs js/Error) _
+                                         {:message "Scan Error"})))))
+                        not-empty)]
+         (do
+           (attachment/delete-attachments! attachments)
+           (assoc (chain/terminate ctx)
+                  :response
+                  {:status 400
+                   :body {:error
+                          {:message
+                           (format "Attachment scan failed, Errors: %s"
+                                   (cs/join ", "
+                                            (map :message attachment-errors)))}}}))
```

Comment:
I feel like there should be a bit of reformatting in the `scan-attachments` function so that you don't get giant indentations like you see here (at least those exceeding 80 columns).

Reply:
db17bfd4a92a4d8f859a4180457dd9a2cd4b74b3